util.c

```diff
@@ -0,0 +1,15 @@
+#include <stdlib.h>
+#include "util.h"
+
+///Get a random integer in the range [lo, hi).
+int random_i(int lo, int hi){
+    return (rand() / (RAND_MAX / (hi - lo) + 1)) + lo;
```

Comment:
This doesn't look right, did you mean, I think you miss a `()`:

```c
    return (rand() / (RAND_MAX /  (hi - lo) + 1) ) + lo;
    return (rand() / (RAND_MAX / ((hi - lo) + 1))) + lo;
```

Reply:
It was roughly correct as written, except for the obvious division by zero when `lo == hi`. E.g. if `random_i(0, N)` returns the maximum of `RAND_MAX / ((RAND_MAX / N) + 1)`, which is `< N`. 

I've altered this in 7d2e17e to give less surprising behaviour. It's still slightly biased when `hi - lo` is close to `RAND_MAX` (see http://c-faq.com/lib/randrange.html), but it's not important for our use.